File: src/frontend/src/types/message.ts
interface Message {
  id: string
  memberId: string
  type: MessageType
  contents: { text: string }
  createdAt?: string
  updatedAt?: string
}

interface ChannelMessage {
  id?: number // RESPONSE에만 있음
  chatType?: 'CHANNEL' | 'DM' // 과거메세지 받을 땐 없음
  serverId: number
  channelId: number
  sequence?: number // RESPONSE랑 과거 메세지 조회에만 있음
  sendMemberId: number
  content: string
  createdAt?: string
}

interface SearchMessage {
  id: number
  serverId: number
  sequence: number
  channelId: number
  sendMemberId: number
  content: string
  createdAt: string
}

type MessageType = 'CHANNEL' | 'DM'

export { type ChannelMessage, type Message, type MessageType, type SearchMessage }

File: src/android/bbebig-discod-clone/presentation/src/main/java/com/smilegate/bbebig/presentation/ui/home/model/Message.kt
package com.smilegate.bbebig.presentation.ui.home.model

import com.smilegate.bbebig.domain.model.MessageDomainModel

data class Message(
    val attachedFiles: List<AttachedFileModel>,
    val channelId: Long,
    val content: String,
    val createdAt: String,
    val deleted: Boolean,
    val id: Long,
    val messageType: String,
    val sendMemberId: Long,
    val sequence: Long,
    val serverId: Long,
    val updatedAt: String,
)

fun MessageDomainModel.toUiModel(): Message {
    return Message(
        attachedFiles = attachedFiles.map { it.toUiModel() },
        channelId = channelId,
        content = content,
        createdAt = createdAt,
        deleted = deleted,
        id = id,
        messageType = messageType,
        sendMemberId = sendMemberId,
        sequence = sequence,
        serverId = serverId,
        updatedAt = updatedAt,
    )
}

fun ChatModel.toMessage(): Message {
    return Message(
        attachedFiles = attachedFiles ?: emptyList(),
        channelId = channelId,
        content = content,
        createdAt = createdAt.orEmpty(),
        deleted = false,
        id = id ?: -1,
        messageType = messageType,
        sendMemberId = sendMemberId,
        sequence = sequence ?: -1,
        serverId = serverId,
        updatedAt = updatedAt.orEmpty(),
    )
}
